src/materialized/bin/materialized.rs

```diff
@@ -151,7 +146,24 @@ fn run() -> Result<(), failure::Error> {
 
     let log_file = popts.opt_str("log-file");
     let max_increment_ts_size = popts.opt_get_default("batch-size", 10000_i64)?;
-    let threads = popts.opt_get_default("threads", 1)?;
+
+    let threads = match popts.opt_get::<usize>("threads")? {
+        Some(val) => val,
+        None => match env::var_os("MZ_THREADS") {
+            Some(val) => val.into_string().unwrap().parse()?,
```

Comment:
Unwrapping here will panic if `MZ_THREADS` contains non-Unicode characters. Instead you'll want to use `env::var("MZ_THREADS")`, and match on the error, ignoring `VarError::NotPresent` and propagating `VarError::NotUnicode`.

Reply:
Great catch!